### ISSUE TITLE
spconv with torch.cuda.amp.autocast

### ISSUE THREAD

**Issue by abhinavagarwalla** [traveller59/spconv#316]
Hi!
I was trying to train a sparse convolutions based model using mixed precision and apex.
```
with torch.cuda.amp.autocast():
    outputs = model(features, batch_size)
```
The model uses `spconv.SubMConv3D`.
Error:
```
Exception has occurred: RuntimeError
expected scalar type Half but found Float
```

The input is a SparseConvTensor with SparseConvTensor.features as a HalfTensor. The error seems to be from [here](https://github.com/traveller59/spconv/blob/fad3000249d27ca918f2655ff73c41f39b0f3127/spconv/conv.py#L196) as the self.weight is not cast to torch.cuda.HalfTensor. 

I also tried using apex. 'O2' opt_level which explicitly casts model weights to fp16 enables training but the loss is nan. 'O1' on the other hand results in the same error as above.
I am not sure if this is spconv specific issue. Let me know in case I should open an issue in pytorch instead.

**Comment by NiccoloCavagnero** [comment]
Same here, the spconv.SubMConv3D is not cast to the proper dtype.
